pkg/vm/stackitem/json.go

```diff
@@ -230,6 +241,11 @@ func (d *decoder) decodeMap() (*Map, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		d.count--
+		if d.count < 0 {
+			return nil, errTooBigElements
+		}
 		k, ok := key.(string)
 		if !ok {
 			return m, nil
```

Comment:
If the key is not a string, then is it a delimiter? And if so, don't we need to increment the count before return, i.e.:
```
k, ok := key.(string)
if !ok {
        d.count++
	return m, nil
}
```
Because these five lines above are for the map key which is supposed to be a string.

Or don't we want to swap these two blocks of code?
```
k, ok := key.(string)
if !ok {
	return m, nil
}
d.count--
if d.count < 0 {
	return nil, errTooBigElements
}
```

Reply:
Fixed, added a test for this.